internal/gitea/team.go

```diff
@@ -0,0 +1,210 @@
+package gitea
+
+import (
+	"fmt"
+	"log"
+	"slices"
+	"strings"
+
+	"code.gitea.io/sdk/gitea"
+	"github.com/jnsgruk/releasegen/internal/repos"
+)
+
+const reposPerPage = 25 // Get this many repos from the API at once.
+const maxPages = 100    // Give up after getting this many pages.
+
+type repoMeta struct {
+	repo            *gitea.Repository
+	monorepoSources []string
+}
+
+// publicOrgRepos finds repositories that are public, not archived, and belong
+// to the specified organisation.
+func publicOrgRepos(org OrgConfig, client *gitea.Client) ([]repoMeta, error) {
+	var orgRepos []repoMeta
+
+	currentPage := 1
+	for pageCount := 0; pageCount < maxPages; pageCount += 1 {
```

Comment:
Should we return an error if it hits `maxPages`? You could do this either by checking if `pageCount >= maxPages-1` inside the loop, or moving the initialisation of `pageCount` outside the loop and testing after.

Go style nit: you normally use `++` in Go: `pageCount++` (Python habits die hard).